plugins/services/src/js/components/modals/ServiceResumeModal.js

```diff
@@ -0,0 +1,179 @@
+import {Confirm} from 'reactjs-components';
+import React, {PropTypes} from 'react';
+import PureRender from 'react-addons-pure-render-mixin';
+
+import AppLockedMessage from './AppLockedMessage';
+import FieldInput from '../../../../../../src/js/components/form/FieldInput';
+import FormGroup from '../../../../../../src/js/components/form/FormGroup';
+import FormRow from '../../../../../../src/js/components/form/FormRow';
+import ModalHeading from '../../../../../../src/js/components/modals/ModalHeading';
+import Service from '../../structs/Service';
+import ServiceTree from '../../structs/ServiceTree';
+
+const METHODS_TO_BIND = [
+  'handleConfirmation',
+  'handleInstancesFieldChange'
+];
+
+class ServiceResumeModal extends React.Component {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      instancesFieldValue: null,
+      errorMsg: null
+    };
+
+    this.shouldComponentUpdate = PureRender.shouldComponentUpdate.bind(this);
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentWillUpdate(nextProps) {
+    const requestCompleted = this.props.isPending
+      && !nextProps.isPending;
+    const shouldClose = requestCompleted && !nextProps.errors;
+
+    if (shouldClose) {
+      this.props.onClose();
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const {errors} = nextProps;
+    if (!errors) {
+      this.setState({errorMsg: null});
+
+      return;
+    }
+
+    if (typeof errors === 'string') {
+      this.setState({errorMsg: errors});
+
+      return;
+    }
+
+    let {message: errorMsg = '', details} = errors;
+    const hasDetails = details && details.length !== 0;
+
+    if (hasDetails) {
+      errorMsg = details.reduce(function (memo, error) {
+        return `${memo} ${error.errors.join(' ')}`;
+      }, '');
+    }
+
+    if (!errorMsg || !errorMsg.length) {
+      errorMsg = null;
+    }
+
+    this.setState({errorMsg});
+  }
+
+  handleConfirmation() {
+    this.props.resumeService(
+      this.state.instancesFieldValue || 0,
```

Comment:
Perhaps you mean `|| 1` ? Otherwise when the `MARATHON_SINGLE_INSTANCE_APP` is present it tries to scale it to `0`